### PR TITLE
Add accessors for enzyme associated data

### PR DESCRIPTION
**Pull request by stelmo**

### CODE REVIEW

**Comment by stelmo** [comment]
@exaexa could you look at the accessors I added? I will fix the tests soon

File: src/types/accessors/AbstractMetabolicModel.jl [hunk 1]
```diff
@@ -369,3 +369,33 @@ By default, it should be safe to do nothing.
 function precache!(a::AbstractMetabolicModel)::Nothing
     nothing
 end
+
+"""
+$(TYPEDSIGNATURES)
+
+Return the molar mass of translated gene with ID `gid`.
+"""
+gene_protein_molar_mass(model::AbstractMetabolicModel, gid::String)::Maybe{Float64} = nothing
```
Comment (exaexa) [inline src/types/accessors/AbstractMetabolicModel.jl:378]: this is normally called a "gene product", might be useful to stay with that.

File: src/types/accessors/AbstractMetabolicModel.jl [hunk 2]
```diff
@@ -369,3 +369,33 @@ By default, it should be safe to do nothing.
...
+the backward direction.
+"""
+reaction_kcat_backward(model::AbstractMetabolicModel, rid::String)::Maybe{Float64} = nothing
+
+"""
+$(TYPEDSIGNATURES)
+
+Return all the [`Isozyme`](@ref)s associated with a `model` and reaction `rid`.
+"""
+reaction_isozymes(model::AbstractMetabolicModel, rid::String)::Maybe{Vector{Isozyme}} = nothing
```
Comment (exaexa) [inline src/types/accessors/AbstractMetabolicModel.jl:401]: This is not really orthogonal to the reaction_kcat_forward and reaction_kcat_backward above. I'd vote for removing these two instead.

Also, any reason why the enzyme (isozyme/geneproduct) mass can't be saved directly in `struct Isozyme` and has to be separated?

In short, we should choose between having the accessors to the tiniest bits or the accessors for sensible units (enzymes, as with isozymes). I'm more in favor of the latter.

Comment (exaexa) [reply src/types/accessors/AbstractMetabolicModel.jl:401]: (on the second read I think I got it -- `reaction_kcat_xxx` is the simplified view for the simplified enzyme constraints. Let's instead do a wrapper that smashes the isozymes? That could also finally kill the (nonsential) duality between gecko and smoment impls.)

File: src/utils/enzyme_constrained.jl [hunk 3]
```diff
@@ -55,37 +55,3 @@ A pipe-able variant of [`gene_product_mass`](@ref).
...
-gene_mass_function = gid -> 1.234
-
-best_isozyme_for_simplified_enzyme_constrained = argmax(
-    simplified_enzyme_constrained_isozyme_speed(gene_mass_function),
-    my_isozyme_vector,
-)
-```
-"""
-simplified_enzyme_constrained_isozyme_speed(gene_product_molar_mass::Function) =
-    isozyme -> simplified_enzyme_constrained_isozyme_speed(isozyme, gene_product_molar_mass)
```
Comment (exaexa) [inline src/utils/enzyme_constrained.jl:91]: this is killed with no replacement?

**Review by exaexa — COMMENTED** [review]
looks okay, some structural cleaning needed

File: src/utils/enzyme_constrained.jl (hunk 3 above)
Comment (stelmo) [reply src/utils/enzyme_constrained.jl:91]: this is now directly in the smoment constructor

File: src/types/accessors/AbstractMetabolicModel.jl (hunk 1 above)
Comment (stelmo) [reply src/types/accessors/AbstractMetabolicModel.jl:378]: fixed

**Review by exaexa — APPROVED** [review]
LGTM, FFTM

**Pull request merged**